### DISCUSSION TITLE
Working on a core integration

### DISCUSSION THREAD

**Discussion by chemelli74** [gicamm/homeassistant-comelit#50] [General]
Hi everyone,

I'm working on a core integration for Comelit.
Especially because I found a Serial Bridge in the rent apartment and not a hub.

What about working all together as a Team to get it feature complete and merged in core ?

First PR is here: https://github.com/home-assistant/core/pull/96552

If you want do discuss, please join HomeAssistant  on [Discord](https://discord.com/invite/home-assistant) and then contact me: `chemelli74#2180`

Simone

**Comment by gicamm** [comment]
Hi Simone,
Currently, the integration is available on HACS.
Here everyone, as happened, makes changes/improvements/fixes.
So the only question could be to move the code to the core.
That's something we can discuss.

Cheers
G

**Comment by chemelli74** [comment]
Hi,

thx for feedback.
There are a few things to change to get it in like for example make it async, move low level device code to a library.

I worked out a library for the Serial Bridge and is working fine: https://github.com/chemelli74/aiocomelit/
Maybe you would like to join the project and add support for HUB and help completing support for VEDO.

Once the library is feature complete, then we can work on the integration side.
As per above, I did a first commit (new integration needs to be as small as possible, so only lights for now).
Once in, it would be easy to add more feature.

I hope you would like to be listed as codeowner.

Simone

P.S.> If you want we can easily chat if you join HomeAssistant  on [Discord](https://discord.com/invite/home-assistant) and then contact me: `chemelli74#2180`

Simone

**Reply by chemelli74** [reply]
Hi all,

For your information, the first domain (light) of the first device (Serial Bridge) is now part of the core.

Hopefully, you will all join the effort to make it feature complete.

Simone

**Reply by gicamm** [reply]
Hi Simone,
great news.

**Reply by chemelli74** [reply]
Hi,

Hopefully you are ready to contribute 😉

Simone
